pkg/api/testing/fuzzer.go

```diff
@@ -429,6 +430,13 @@ func FuzzerFor(t *testing.T, version unversioned.GroupVersion, src rand.Source)
 				}
 			}
 		},
+		func(r *runtime.RawExtension, c fuzz.Continue) {
+			types := []runtime.Object{&api.Pod{}, &extensions.Deployment{}, &api.Service{}}
+			t := types[c.Rand.Intn(len(types))]
+			c.Fuzz(t)
+			by, _ := json.Marshal(t)
```

Comment:
Yes, and you shouldn't use json, you should use runtime.Encode. Raw JSON encoding won't set the api version & kind on the wire, which will make the object uninterpretable.